bin/common.c

```diff
@@ -75,3 +78,57 @@ int key_log_callback(void *file, struct s2n_connection *conn, uint8_t *logline,
 
     return fflush((FILE *)file);
 }
+
+static int s2n_get_psk_hmac_alg(s2n_psk_hmac *psk_hmac, char *hmac_str)
+{
+    POSIX_ENSURE_REF(psk_hmac);
+    POSIX_ENSURE_REF(hmac_str);
+
+    if (strcmp(hmac_str, "S2N_PSK_HMAC_SHA256") == 0) {
+        *psk_hmac = S2N_PSK_HMAC_SHA256;
+    } else if (strcmp(hmac_str, "S2N_PSK_HMAC_SHA384") == 0) {
+        *psk_hmac = S2N_PSK_HMAC_SHA384;
+    } else {
+        return S2N_FAILURE;
+    }
+    return S2N_SUCCESS;
+}
+
+int s2n_setup_external_psk(struct s2n_psk *psk_list[S2N_MAX_PSK_LIST_LENGTH], size_t *psk_idx, char *params)
+{
+    POSIX_ENSURE_REF(psk_list);
+    POSIX_ENSURE_REF(psk_idx);
+    POSIX_ENSURE_REF(params);
+
+    struct s2n_psk *psk = s2n_external_psk_new();
+    POSIX_ENSURE_REF(psk);
+    POSIX_GUARD_RESULT(s2n_psk_init(psk, S2N_PSK_TYPE_EXTERNAL));
+    /* Default HMAC algorithm is S2N_PSK_HMAC_SHA256 */
```

Comment:
Is s2n_psk_init a public symbol? Do you need to do it after calling s2n_external_psk_new?